src/algo/depth-to-rgb-calibration/optimizer.cpp

```diff
@@ -435,7 +435,118 @@ void grid_xy(
+{
+    std::vector<uint8_t> is_supressed;
+    auto loc_edg_it = local_edges.begin();
+    for (auto i = 0; i < valid_size; i++)
+    {
+        //isSupressed = localEdges(:,3) >= localEdges(:,2) & localEdges(:,3) >= localEdges(:,4);
+        auto vec2 = *(loc_edg_it + 1);
+        auto vec3 = *(loc_edg_it + 2);
+        auto vec4 = *(loc_edg_it + 3);
+        loc_edg_it += 4;
+        bool res = (vec3 >= vec2) && (vec3 >= vec4);
+        is_supressed.push_back(res);
+    }
+    return is_supressed;
+}
+std::vector<double> depth_mean(std::vector<double>& local_x, std::vector<double>& local_y)
+{
+    std::vector<double> res;
+    size_t size = local_x.size() / 2;
+    auto itx = local_x.begin();
+    auto ity = local_y.begin();
+    for (auto i = 0; i < size; i++, ity += 2, itx += 2)
+    {
+        double valy = (*ity + *(ity + 1)) / 2;
+        double valx = (*itx + *(itx + 1)) / 2;
+        res.push_back(valy);
+        res.push_back(valx);
+    }
+
+    return res;
+}
+std::vector<double> sum_gradient_depth(std::vector<double> &gradient, std::vector<double> &direction_per_pixel)
+{
+    std::vector<double> res;
+    size_t size = direction_per_pixel.size() / 2;
+    auto it_dir = direction_per_pixel.begin();
+    auto it_grad = gradient.begin();
+    for (auto i = 0; i < size; i++, it_dir+=2, it_grad+=2)
+    {
+        // normalize : res = val/sqrt(row_sum)
+        auto rorm_dir1 = *it_dir / sqrt(abs(*it_dir) + abs(*(it_dir + 1)));
+        auto rorm_dir2 = *(it_dir+1) / sqrt(abs(*it_dir) + abs(*(it_dir + 1)));
+        auto val = abs(*it_grad * rorm_dir1 + *(it_grad + 1) * rorm_dir2);
+        res.push_back(val);
+    }
+    return res;
+}
+std::vector< double > find_valid_depth_edges(std::vector<double> grad_in_direction, std::vector<uint8_t> is_supressed, std::vector<double>values_for_subedges, int threshold)
+{
+    std::vector< double > res;
+    //validEdgePixels = zGradInDirection > params.gradZTh & isSupressed & zValuesForSubEdges > 0;
+    for (int i = 0; i < grad_in_direction.size(); i++)
+    {
+        bool cond1 = *(grad_in_direction.begin() + i) > threshold;
+        bool cond2 = *(is_supressed.begin() + i) == true;
+        bool cond3 = *(values_for_subedges.begin() + i) > 0;
+        if (cond1 && cond2 && cond3)
+        {
+            res.push_back(1);
+        }
+        else
+        {
+            res.push_back(0);
+        }
+    }
+    return res;
+}
+std::vector<double> find_local_values_min(std::vector<double>& local_values)
+{
+    std::vector<double> res;
+    size_t size = local_values.size() / 4;
+    auto it = local_values.begin();
+    for (auto i = 0; i < size; i++)
+    {
+        auto val1 = *it;
+        auto val2 = *(it + 1);
+        auto val3 = *(it + 2);
+        auto val4 = *(it + 3);
+        it += 4;
+        double res_val = std::min(val1, std::min(val2, std::min(val3, val4)));
+        res.push_back(res_val);
+    }
+    return res;
+}
 void optimizer::set_depth_data(
```

Comment:
You'll have to explain it to me: why are we setting the ir_data in set_depth_data? Did you remove set_ir_data()?

Reply:
Also, if we're keeping set_ir_data(), then I'd like to keep the APIs with the same names as before, i.e. set_z_data()